testament/testament.nim

```diff
@@ -1029,7 +1262,7 @@ func defaultTargets(category: Category): set[TTarget] =
   else:
     standardTargets
 
-proc testSpec(r: var TResults, test: TTest) =
+proc testSpec(r: var TResults, test: TTest, execution: Execution) =
```

Comment:
Not sure what I was thinking weaving through another parameter ('execution') rather than combining it with existing ones. Sigh... sorry.

Reply:
should I merge these two? I had a feeling they are effectively the same (state object) but wasn't exactly sure. 

There are a few moments still where I need to bring in `TResults` in as well

Reply:
I could be really tired and that's yet another issue...

My sleepy brain is thinking (or not) that Execution makes sense as a first parameter. Then a second one to identify the test makes sense and that should be immutable data. Then finally result which can be separate to read only and writeable data. Issue with dod is when you just get this super blob that any proc can do anything to at any time.